content/blog/2016/2016-05-19-update-plugin-for-pipeline-tutorial.md

```diff
@@ -0,0 +1,189 @@
+
+Then, I moved on to [actually creating the step](https://github.com/cprice404/gatling-plugin/commit/3de3485be591c7b750ec2671e74558a79efc4319).  The meat of the code was this:
+
+```java
+public class GatlingArchiverStep extends AbstractStepImpl {
+    @DataBoundConstructor
+    public GatlingArchiverStep() {}
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+        public DescriptorImpl() { super(GatlingArchiverStepExecution.class); }
+
+        @Override
+        public String getFunctionName() {
+            return "gatlingArchive";
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return "Archive Gatling reports";
+        }
+    }
+}
+```
+
+Note that in that commit I also added a `config.jelly` file.  This is how you define the UI for your step, which will show up in the Snippet Generator.  In the case of this Gatling step there's really not much to configure, so my `config.jelly` is basically empty.
+
+With that (and the rest of the code from that commit) in place, I was able to fire up the development Jenkins server (via `mvn hpi:run`, and note that you need to go into the "Manage Plugins" screen on your development server and install the Pipeline plugin once before any of this will work) and visit the Snippet Generator to see if my step showed up in the dropdown:
+
+![gatlingArchive snippet](images/2016-05-19-update-plugin-for-pipeline-tutorial/10_snippet_generator.png)
+
+GREAT SUCCESS!
+
+This step doesn't actually *do* anything yet, but it's recognized by Jenkins and can be included in your pipeline scripts at that point, so, we're on our way, kids!
+
+### The `step` metastep
+
+The step that we created above is a first-class DSL addition that can be used in Pipeline scripts.  There's another way to make your plugin work usable from a Pipeline job, without making it a first-class build step.  This is by use of the `step` "metastep", mentioned in the [pipeline-plugin DEVGUIDE.md](https://github.com/jenkinsci/pipeline-plugin/blob/893e3484a25289c59567c6724f7ce19e3d23c6ee/DEVGUIDE.md#build-steps).  When using this approach, you simply refactor your `Builder` or `Publisher` to extend `SimpleBuildStep`, and then you can reference the build step from the Pipeline DSL using the `step` method.
+
+Whoa, that's a lot of uses of the word "step".  Let's back up a... umm.. step.
```

Comment:
This is definitely too "familiar" for me. Just refactor to say "step" fewer times, or ignore it.